core/src/main/scala/org/apache/spark/Namespace.scala

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark
+
+/** A simple trait that manages namespaces in Spark configuration properties and env variables */
+sealed private[spark] trait Namespace {
+  /** Namespace name for env variables */
+  def envNamespace: String
+
+  /** Namespace name for conf property */
+  def confNamespace: String
+
+  /** Adds this namespace to the given env variable name. Assuming the given `envKey` is
```

Comment:
this doesn't follow the multi-line scaladoc format used in Spark.

    /**
     * Text.
     */

Same in other places.